### PR TITLE
App Insights implemented for frontend & backend

### PR DESCRIPTION
**Pull request by michaeldeongreen**
- App Insights configured on backend
- App Insights configured on the frontend
- Distributed Tracing configured
- Updated documentation

### CODE REVIEW

File: api/src/LinkyLink/Startup.cs [hunk 1]
```diff
@@ -85,6 +80,9 @@ public void ConfigureServices(IServiceCollection services)
                 var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
                 c.IncludeXmlComments(xmlPath);
             });
+            
+            // Add Application Insights Telemetry
```
Comment (Humoiz) [inline api/src/LinkyLink/Startup.cs]: I have a PR which adds similar code. I would prefer to have this line commented out. User can uncomment the line to enable application insights if they wish to use it.

Comment (aluong) [reply api/src/LinkyLink/Startup.cs]: Is there a reason why we just don't want it on all the time?

File: frontend/src/App.vue [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { Component, Prop, Vue } from "vue-property-decorator";
 import NavBar from "@/components/NavBar.vue";
 import ProgressBar from "@/components/ProgressBar.vue";
 
+
```
Comment (aluong) [inline frontend/src/App.vue]: Do you need this extra space for linting?

File: api/src/LinkyLink/Startup.cs (hunk 1 above)
Comment (Humoiz) [reply api/src/LinkyLink/Startup.cs]: Almost everything will be wired up with App insights but I was thinking may be attendees should enable this if they choose to go with App Insights. Anyway, I dont have a strong preference.

Comment (aluong) [reply api/src/LinkyLink/Startup.cs]: I prefer leaving it there. Not really our intention to show you how to add App Insights to your app right? It's more like use it answer your business problems, or if you don't like it hook up something else. 

Go for it, swap in FluentD and send logs into Grafana.

Comment (michaeldeongreen) [reply api/src/LinkyLink/Startup.cs]: Let's discuss after Stand Up.  If we believe it should be off on the backend, then we need to do the same for the frontend as it doesn't make sense to collect telemetry by default on the frontend and not the backend, especially with distributed tracing enabled by default.

I personally believe that it should be on by default.  It goes back to the "who would deploy an app to production with no monitoring?".  The narrative states that we were in a hurry to go live but we at least had "some" monitoring but it was not finely tuned.

File: frontend/src/App.vue (hunk 2 above)
Comment (michaeldeongreen) [reply frontend/src/App.vue]: Updated.

File: api/src/LinkyLink/Startup.cs (hunk 1 above)
Comment (Humoiz) [reply api/src/LinkyLink/Startup.cs]: I agree with both of you. Lets keep it on by default. I`ll change my other PR to keep it on. as well.

**Pull request merged**